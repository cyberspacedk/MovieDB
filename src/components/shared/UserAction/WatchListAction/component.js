import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'antd';

const WatchListAction = ({ handleWatchList, watchListStatus }) => {
  return (
    <Icon
      type="book"
      theme={watchListStatus ? 'filled' : undefined}
      onClick={handleWatchList}
    />
  );
};

WatchListAction.propTypes = {
  handleWatchList: PropTypes.func.isRequired,
  watchListStatus: PropTypes.bool.isRequired,
};

export default WatchListAction;
