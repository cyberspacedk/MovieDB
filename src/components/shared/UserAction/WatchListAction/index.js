import WatchListActionContainer from './container';

export default WatchListActionContainer;
