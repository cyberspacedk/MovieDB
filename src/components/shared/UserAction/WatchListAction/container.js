/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import WatchListAction from './component';
import { operationsWatchListRequest } from '../../../../store/watchList/actions';

class WatchListActionContainer extends Component {
  state = {
    watchListStatus: false,
  };

  handleWatchList = () => {
    this.setState(prevState => ({
      watchListStatus: !prevState.watchListStatus,
    }));
    this.props.operationsWatchListRequest(this.props.movieId, true);
  };

  render() {
    const { watchListStatus } = this.state;
    return (
      <WatchListAction
        handleWatchList={this.handleWatchList}
        watchListStatus={watchListStatus}
      />
    );
  }
}

const mdtp = {
  operationsWatchListRequest,
};

export default connect(
  null,
  mdtp,
)(WatchListActionContainer);
