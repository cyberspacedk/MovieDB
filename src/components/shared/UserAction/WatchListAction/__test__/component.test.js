import React from 'react';
import { shallow } from 'enzyme';
import WatchListAction from '../component';

describe('Component: WatchListAction', () => {
  const props = {
    handleWatchList: jest.fn(),
    watchListStatus: false,
  };

  const wrapper = shallow(<WatchListAction {...props} />);

  it('Match its snapshot.', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
