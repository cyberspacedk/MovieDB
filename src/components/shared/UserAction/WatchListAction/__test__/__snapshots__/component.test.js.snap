// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: WatchListAction Match its snapshot. 1`] = `
<Icon
  onClick={[MockFunction]}
  type="book"
/>
`;
